nyxx_interactions/lib/src/internal/sync/LockFileCommandSync.dart

```diff
@@ -0,0 +1,123 @@
+part of nyxx_interactions;
+
+/// Manually define command syncing rules
+class LockFileCommandSync implements ICommandsSync {
+  const LockFileCommandSync();
+
+  @override
+  FutureOr<bool> shouldSync(Iterable<SlashCommandBuilder> commands) async {
+    final lockFile = File("./nyxx_interactions.lock");
+    final lockFileMapData = <String, String>{};
+
+    commands.forEach(
+      (c) {
+        lockFileMapData[c.name] = new _LockfileCommand(
+          c.name,
+          c.description,
+          c.guild,
+          c.defaultPermissions,
+          c.permissions?.map((p) => new _LockfilePermission(p._type, p.id, p.hasPermission)) ?? [],
+          c.options.map((o) => new _LockfileOption(o.type.value, o.name, o.description, o.options ?? [])),
+        ).generateHash();
+      },
+    );
```

Comment:
cant be for (final command in commands) be used?

Reply:
Yup that works too. Changes done 👍🏼